core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -201,9 +277,43 @@ private static Method getParentMethod(String className, String methodName) {
 		}
 	}
 
-	public FrankDocGroup addGroup(String name) {
-		FrankDocGroup group = new FrankDocGroup(name);
-		groups.add(group);
-		return group;
+	private List<ConfigChild> createConfigChildren(	Method[] methods, FrankElement parent) {
+		List<Method> configChildSetters = Arrays.asList(methods).stream()
+				.filter(Utils::isConfigChildSetter)
+				.filter(m -> configChildDescriptors.get(m.getName()) != null)
+				.collect(Collectors.toList());
+		List<ConfigChild> result = new ArrayList<>();
+		for(Method m: configChildSetters) {
+			ConfigChild configChild = new ConfigChild(parent);
+			ConfigChildSetterDescriptor configChildDescriptor = configChildDescriptors.get(m.getName());
+			IbisDoc ibisDoc = AnnotationUtils.findAnnotation(m, IbisDoc.class);
+			Integer optionalOrder = parseIbisDocAnnotation(ibisDoc);
+			if(optionalOrder != null) {
+				configChild.setSequenceInConfig(optionalOrder);
+			} else {
+				log.warn(String.format("No config child order for method [%s] of Frank element [%s]",
+						m.getName(), parent.getSimpleName()));
+				configChild.setSequenceInConfig(Integer.MAX_VALUE);
+			}
+			Class<?> elementClass = m.getParameterTypes()[0];
+			configChild.setElementType(findOrCreateElementType(elementClass));
+			configChild.setAllowMultiple(configChildDescriptor.isAllowMultiple());
+			configChild.setMandatory(configChildDescriptor.isMandatory());
+			configChild.setSyntax1Name(configChildDescriptor.getSyntax1Name());
+			result.add(configChild);
+		}
+		return result;
+	}
+
+	private static Integer parseIbisDocAnnotation(IbisDoc ibisDoc) {
```

Comment:
Deze parseIbisDocAnnotation() is er alleen voor ConfigChildren. Kan hij dan niet daar een method van zijn?

Reply:
Hier hebben we het daarnet over gehad door de telefoon. Om attributes te creëren moet je @IbisDocRef annotaties parsen en de betreffende FrankElement objecten creëren. Dat zit weer in het model. Je voorstel zou betekenen dat FrankDocModel de class FrankAttribute moet importeren en ook omgekeerd. Dat lukt niet.

Hetzelfde geldt voor configuration children. Om een ConfigChild te maken moet je het ElementType creëren dat erin gaat.

Reply:
Oh sorry, verkeerd gekeken. Dit punt kan ik wèl meenemen.

Reply:
Gedaan.